kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -204,6 +215,9 @@ public void put(Collection<SinkRecord> records) throws ConnectException {
       TopicPartition tp = new TopicPartition(topic, partition);
 
       if (maybeSkipOnNullValue(record)) {
+        if (reporter != null) {
+          reporter.report(record, new ConnectException("Cannot write null value record"));
```

Comment:
Should this be `DataException`?

Reply:
I believe so. This would match the other use case right?
Have you checked how the exception is used by the reporter?

Reply:
Yes, this would match the other cases, updated.